org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/testexecution/TestProcess.xtend

```diff
@@ -0,0 +1,62 @@
+package org.testeditor.web.backend.testexecution
+
+import static org.testeditor.web.backend.testexecution.TestStatus.*
+
+class TestProcess {
+
+	public static val DEFAULT_IDLE_TEST_PROCESS = new TestProcess()
+	public static val WAIT_TIMEOUT_SECONDS = 5
+
+	var Process process
```

Comment:
Well, I never actually write anything on Process instances. The only write is "nulling out" the class's reference to a process, which is why I copy the reference into a local variable every time, because when the process terminates, it might take away the reference right in the middle of another thread's way through the method.

I'm a bit worried that synchronizing (too extensively) might lead to potential deadlock scenarios, which are easily overlooked, and are basically impossible to test against…

Reply:
… added my reasoning of why I believe the class should be thread-safe as a class-level JavaDoc comment.